src/com/amazon/ion/impl/lite/IonSymbolLite.java

```diff
@@ -101,28 +101,26 @@ && _stringValue() == null) {
     }
 
     @Override
-    int hashCode(SymbolTableProvider symbolTableProvider)
+    int hashSignature() {
+        return HASH_SIGNATURE;
+    }
+
+    @Override
+    int scalarHashCode()
     {
         final int sidHashSalt   = 127;      // prime to salt sid
         final int textHashSalt  = 31;       // prime to salt text
         int result = HASH_SIGNATURE;
+        int tokenHashCode = _text_value == null
+            ? _sid  * sidHashSalt
+            : _text_value.hashCode() * textHashSalt;
 
-        if (!isNullValue())
-        {
-            SymbolToken token = symbolValue(symbolTableProvider);
-            String text = token.getText();
-
-            int tokenHashCode = text == null
-                ? token.getSid()  * sidHashSalt
-                : text.hashCode() * textHashSalt;
+        // mixing to account for small text and sid deltas
+        tokenHashCode ^= (tokenHashCode << 29) ^ (tokenHashCode >> 3);
 
-            // mixing to account for small text and sid deltas
-            tokenHashCode ^= (tokenHashCode << 29) ^ (tokenHashCode >> 3);
+        result ^= tokenHashCode;
```

Comment:
All of this still makes me suspicious of our hashing algorithms, as described earlier. I'd rather use [`Arrays.hashCode(Object... a)`](https://docs.oracle.com/javase/8/docs/api/java/util/Arrays.html#hashCode-java.lang.Object:A-) and count on ordering of the parts to give us distinct hashes values with traded `_sid` and `_text_value.hashCode()`.

This looks correct to me, but I'd rather own as little code as possible. I don't see what we win out of having our own implementations. That's what I'm missing.

Reply:
See my comment here: https://github.com/amazon-ion/ion-java/pull/546#discussion_r1295123192